src/librustc_mir/transform/const_prop.rs

```diff
@@ -257,86 +257,19 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
-                    | StackFrameLimitReached
-                    | NoMirFor(..)
-                    | InlineAsm
-                    => {},
-
-                    | InvalidMemoryAccess
-                    | DanglingPointerDeref
-                    | DoubleFree
-                    | InvalidFunctionPointer
-                    | InvalidBool
-                    | InvalidDiscriminant(..)
-                    | PointerOutOfBounds { .. }
-                    | InvalidNullPointerUsage
-                    | ValidationFailure(..)
-                    | InvalidPointerMath
-                    | ReadUndefBytes(_)
-                    | DeadLocal
-                    | InvalidBoolOp(_)
-                    | DerefFunctionPointer
-                    | ExecuteMemory
-                    | Intrinsic(..)
-                    | InvalidChar(..)
-                    | AbiViolation(_)
-                    | AlignmentCheckFailed{..}
-                    | CalledClosureAsFunction
-                    | VtableForArgumentlessMethod
-                    | ModifiedConstantMemory
-                    | ModifiedStatic
-                    | AssumptionNotHeld
-                    // FIXME: should probably be removed and turned into a bug! call
-                    | TypeNotPrimitive(_)
-                    | ReallocatedWrongMemoryKind(_, _)
-                    | DeallocatedWrongMemoryKind(_, _)
-                    | ReallocateNonBasePtr
-                    | DeallocateNonBasePtr
-                    | IncorrectAllocationInformation(..)
-                    | UnterminatedCString(_)
-                    | HeapAllocZeroBytes
-                    | HeapAllocNonPowerOfTwoAlignment(_)
-                    | Unreachable
-                    | ReadFromReturnPointer
-                    | ReferencedConstant
-                    | InfiniteLoop
-                    => {
-                        // FIXME: report UB here
-                    },
+                    Exit(_) => {},
 
-                    | OutOfTls
-                    | TlsOutOfBounds
-                    | PathNotFound(_)
+                    | Unsupported(OutOfTls)
+                    | Unsupported(TlsOutOfBounds)
+                    | Unsupported(PathNotFound(_))
                     => bug!("these should not be in rustc, but in miri's machine errors"),
 
-                    | Layout(_)
-                    | UnimplementedTraitSelection
-                    | TypeckError
-                    | TooGeneric
-                    // these are just noise
-                    => {},
-
-                    // non deterministic
-                    | ReadPointerAsBytes
-                    // FIXME: implement
-                    => {},
-
+                    | Unsupported(_) => {},
+                    | UndefinedBehaviour(_) => {},
+                    | InvalidProgram(_) => {},
+                    | ResourceExhaustion(_) => {},
                     | Panic(_)
```

Comment:
@oli-obk this was previously a very annoying-to-maintain exhaustive list. I am not sure what to do with it. Does this change seem reasonable to you?

Reply:
I love it. I thought that is the whole point of this change :laughing: